frontend/src/lib/k8s/apiProxy.ts

```diff
@@ -369,16 +380,30 @@ function simpleApiFactoryWithNamespace(
     scale?: ReturnType<typeof apiScaleFactory>;
     [other: string]: any;
   } = {
-    list: (namespace: string, cb: StreamResultsCb, errCb: StreamErrCb) =>
-      streamResults(url(namespace), cb, errCb),
-    get: (namespace: string, name: string, cb: StreamResultsCb, errCb: StreamErrCb) =>
-      streamResult(url(namespace), name, cb, errCb),
-    post: (body: KubeObjectInterface) => post(url(body.metadata.namespace as string), body),
-    patch: (body: OpPatch[], namespace: string, name: string) =>
-      patch(`${url(namespace)}/${name}?pretty=true`, body),
-    put: (body: KubeObjectInterface) =>
-      put(`${url(body.metadata.namespace as string)}/${body.metadata.name}`, body),
-    delete: (namespace: string, name: string) => remove(`${url(namespace)}/${name}`),
+    list: (
+      namespace: string,
+      cb: StreamResultsCb,
+      errCb: StreamErrCb,
+      queryParams?: QueryParameters
+    ) => streamResults(url(namespace), cb, errCb, queryParams),
+    get: (
+      namespace: string,
+      name: string,
+      cb: StreamResultsCb,
+      errCb: StreamErrCb,
+      queryParams?: QueryParameters
+    ) => streamResult(url(namespace), name, cb, errCb, queryParams),
+    post: (body: KubeObjectInterface, queryParams?: QueryParameters) =>
+      post(url(body.metadata.namespace as string) + asQuery(queryParams), body),
```

Comment:
Namespace in metadata is defined as string type. So, I think it would be fine to cast namespace to string. https://github.com/kinvolk/headlamp/blob/f9002aa769caa2ffd00480981b5b51aa27e17069/frontend/src/lib/k8s/cluster.ts#L35

Reply:
Turns out we only use a full (http/s) URL later in the request, so we cannot just call this with `new URL` AFAICS.
I agree with removing the `as string` and passing the assertion instead (`metadata.namespace!`).

Reply:
I don't follow what you mean exactly, but it sounds like you fixed it. Resolving this convo.